### ISSUE TITLE
Metadata streaming announcement: startup announcement is cut off or not announced

### ISSUE THREAD

**Issue by josephsl** [josephsl/stationPlaylist#82]
Hi,

Reported by a broadcaster: if NVDA is told to announce metadata streaming status at startup, it either cuts off or not announced.

Technical: caused by name change event fired by status bar controls, as well as timing; a remote possibility regarding wxPython 4 might also be to blame.

One of the following things can happen:

1. Metadata sound is heard but the actual text isn't.
2. The metadata status text is cut off by status bar announcements.
3. A series of status bar text changes will be announced, followed by metadata streaming announcement.

One possible solution is telling name change event to hold off announcing changes until metadata announcement is finished. Another solution is to delay metadata status announcmenet - say, five seconds instead of two seconds as currently implemented. A more general solution would require speech refactor and message priority. A more subtle solution involves defining an extension point action for Studio window load, which should then allow metadata streaming message to be announced at appropriate time and open other possibilities.

For now, the most feasible solution is delaying metadata announcement, which should allow more important status bar changes to be announced. This takes care of a situation where NVDA restarts while Studio is running, but might not be quite ideal when Studio starts while NVDA is running. A more refined route would be using extension points, but this requires extensive testing and looking for race conditions.

For 18.11 and 18.09.5, delay metadata status announcement; for 18.12 and later (possibly for a later LTS18 release), investigate extension points.

Thanks.

**Comment by josephsl** [comment]
Hi,

Turns out the solution was to keep an eye on initial Studio window as part of foreground event (with an internal flag added).

Thanks.
